src/Workspaces/Remote/ServiceHub/Services/StackTraceExplorer/RemoteStackTraceExplorerService.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.StackTraceExplorer;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+
+namespace Microsoft.CodeAnalysis.Remote.Services.StackTraceExplorer
+{
+    internal sealed class RemoteStackTraceExplorerService : BrokeredServiceBase, IRemoteStackTraceExplorerService
+    {
+        internal sealed class Factory : FactoryBase<IRemoteStackTraceExplorerService>
+        {
+            protected override IRemoteStackTraceExplorerService CreateService(in ServiceConstructionArguments arguments)
+                => new RemoteStackTraceExplorerService(arguments);
+        }
+
+        private int _id;
+
+        public RemoteStackTraceExplorerService(in ServiceConstructionArguments arguments) : base(arguments)
+        {
+        }
+
+        public ValueTask<SerializableDefinitionItem?> TryFindDefinitionAsync(PinnedSolutionInfo solutionInfo, string frameString, StackFrameSymbolPart symbolPart, CancellationToken cancellationToken)
+        {
+            return RunServiceAsync<SerializableDefinitionItem?>(async cancellationToken =>
+            {
+                var solution = await GetSolutionAsync(solutionInfo, cancellationToken).ConfigureAwait(false);
+                if (solution is null)
+                {
+                    throw new InvalidOperationException();
+                }
+
+                var result = await StackTraceAnalyzer.AnalyzeAsync(frameString, cancellationToken).ConfigureAwait(false);
+                if (result.ParsedFrames.Length != 1 || result.ParsedFrames[0] is not ParsedStackFrame parsedFrame)
+                {
+                    throw new InvalidOperationException();
+                }
+
+                var definition = await StackTraceExplorerUtilities.GetDefinitionAsync(solution, parsedFrame.Root, symbolPart, cancellationToken).ConfigureAwait(false);
+                if (definition is null)
+                {
+                    return null;
+                }
+
+                return SerializableDefinitionItem.Dehydrate(_id++, definition);
```

Comment:
```suggestion
                return SerializableDefinitionItem.Dehydrate(id: 0, definition);
```

you don't need unique ids as you only ever create one of these, and you have no references refering to them.

Reply:
Good to know, I'll remove the field and just pass 0